docs/developing.rst

```diff
@@ -0,0 +1,350 @@
+.. highlight:: sh
+
+=================
+Development Guide
+=================
+
+.. rubric:: Table of contents
+
+.. contents::
+   :local:
+   :depth: 2
+
+
+Prerequisites
+=============
+
+JDK
+---
+
+JDK 12 is required to build the plugin because of the dependency on Elasticsearch test framework in our integration test. So you must have a JDK 12 installation on your machine. After the installation, please configure JAVA_HOME environment variable accordingly. If everything goes right, you should be able to see sample output as below on MacOS::
+
+ $ echo $JAVA_HOME
```

Comment:
So I know RST supports any level of indentation, but a single space makes it pretty hard to differentiate the code block from the surrounding text. I recommend three spaces so that you're consistent with the level of spacing that numbered lists require.

```rst
So here's a paragraph.

#. Then we have an autonumbered list.

   Here's a paragraph within a list.
   Note that you need three spaces here.

#. Another entry.

Now let's add a code block::

   Note how this block lines up with my lists.
   I get that three spaces are awkward, so do this however you'd like.

Here's another paragraph.

::

   Here's a code block that uses the alternate style.
   Again, I think three spaces makes the most sense given this syntax.

And a final paragraph.

- Yes, I know that unordered lists.

  Only need two spaces for follow-on paragraphs.

- Oh well.
```

Reply:
Good idea. Will do for all the code blocks.